fedimint-client/src/lib.rs

```diff
@@ -684,6 +684,15 @@ impl Client {
             .find(|(_, kind, _module)| *kind == module_kind)
             .map(|(instance_id, _, _)| instance_id)
     }
+
+    /// Returns the data from which the client's root secret is derived (e.g.
+    /// BIP39 seed phrase struct).
+    pub async fn root_secret_encoding<S>(&self) -> S::Encoding
```

Comment:
Btw, I'm very open to a better name

Reply:
@dpc "root secret encoding" was a bit confusing to me and maybe "encoded secret" or "serialized secret" would be more clear?

Reply:
Well, it's not even encoded really, it's in a form where it can be either converted to a `DerivableSecret` or presumably encoded in a custom way. When you e.g. look at the bip39 impl we have the parsed mnemonic which is the "encoding".

Reply:
@elsirion I see. Maybe "secret key input" or "secret key source"?

I would rename the "strategy" too. "strategy" is the name of the pattern which is useful to describe what it's doing, but doesn't have to be the name of the trait. So `s/RootSecretStrategy/RootSecretInput/g`?

Reply:
The problem is that the struct on which the trait is implemented isn't the "root secret input" itself, but rather a ZST that represents the way some other (associated) type has to be encoded. So in a way the ZST is the strategy.

Reply:
@elsirion Could it be changed to be "root secret input"?

```rust
trait RootSecretInput : Encodable, Decodable {
  fn to_root_secret() -> DeriveableSecret;
  fn random() -> Self;
}
```

I might be missing something, but the only reason to prefer a ZST over trait over the data itself, is if we needed to store somewhere the code without the actual data.

Reply:
It's a choice between needing to wrap everything in newtypes and the ZST hack. All encoding except `PlainRootSecretStrategy` will be implemented by external crates on types of yet other crates (we can't expect `bip39` to be aware of fedimint, it needs a `fedimint-bip39` connector crate). If they needed to impl that trait directly that wouldn't work.

Reply:
@elsirion Is

```rust
struct Bip39Strategy;

impl RootSecretInput {
   type Encoding = bip39::Seedphrase;  

  // ...
}
```

 any better than 

```rust
struct Bip39SecretInput(bip39::Seedphrase);

impl SecretInput {

  // ..
}
```

?

ZST seems about the same, except with the data unnaturally(?) outside.